lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
+            }).catch(function (err: Error) {
                 return reject(err);
             });
         });
     };
 
-    /**
-     * Kill the session by invalidating the access token.
-     * If access_token is passed then it will be set as current access token and get in validated.
-     * @method invalidateAccessToken
-     * @memberOf KiteConnect
-     * @instance
-     * @param {string} [access_token] Token to invalidate. Default is the active `access_token`.
-     */
-    self.invalidateAccessToken = function(access_token) {
-        access_token = access_token || this.access_token;
-
-        return _delete("api.token.invalidate", {
-            api_key: self.api_key,
-            access_token: access_token
+    invalidateAccessToken(access_token?: string): Promise<AxiosResponse<any>> {
+        return this._delete('api.token.invalidate', {
+            api_key: this.api_key,
+            access_token: access_token || this.access_token,
         });
     };
 
-    /**
-     * Renew access token by active refresh token.
-     * Renewed access token is implicitly set.
-     * @method renewAccessToken
-     * @memberOf KiteConnect
-     * @instance
-     *
-     * @param {string} refresh_token Token obtained from previous successful login.
-     * @param {string} api_secret API secret issued with the API key.
-     */
-    self.renewAccessToken = function(refresh_token, api_secret) {
-        return new Promise(function (resolve, reject) {
-            const checksum = sha256(self.api_key + refresh_token + api_secret).toString();
+    renewAccessToken(refresh_token: string, api_secret: string): Promise<AxiosResponse<any>> {
+        return new Promise((resolve, reject) => {
+            const checksum = sha256(this.api_key + refresh_token + api_secret).toString();
 
-            const p = _post("api.token.renew", {
-                api_key: self.api_key,
+            const p = this._post('api.token.renew', {
+                api_key: this.api_key,
                 refresh_token: refresh_token,
                 checksum: checksum
             });
 
-            p.then(function(resp: any) {
+            p.then(function (resp: any) {
                 if (resp && resp.access_token) {
-                    self.setAccessToken(resp.access_token);
+                    this.setAccessToken(resp.access_token);
                 }
                 return resolve(resp);
-            }).catch(function(err) {
+            }).catch(function (err: Error) {
                 return reject(err);
             });
         });
     };
 
-    /**
-     * Invalidate the refresh token.
-     * @method invalidateRefreshToken
-     * @memberOf KiteConnect
-     * @instance
-     * @param {string} refresh_token Token to invalidate.
-     */
-    self.invalidateRefreshToken = function(refresh_token) {
-        return _delete("api.token.invalidate", {
+    invalidateRefreshToken = function (refresh_token: string): Promise<AxiosResponse<any>> {
```

Comment:
Maybe you can follow the same function signature here as well like the rest of all (renewAccessToken, etc) to keep it constant. Suggests maintaining consistency in function signatures across your codebase.
```
invalidateRefreshToken(refresh_token:string):Promise<AxiosResponse<any>> 
{...}
```